Common/Securities/Futures/FutureMarginModel.cs

```diff
@@ -0,0 +1,345 @@
+            }
+        }
+                
+        /// <summary>
+        /// Reads margin requirements file and returns a sorted list of historical margin changes
+        /// </summary>
+        /// <param name="file">The csv file to be read</param>
+        /// <returns>Sorted list of historical margin changes</returns>
+        private MarginRequirementsEntry[] FromCsvFile(string file)
+        {
+            if (!File.Exists(file))
+            {
+                Log.Trace("Unable to locate future margin requirements file. Defaulting to zero margin for this symbol. File: {0}" , file);
+
+                return new[] {
+                                new MarginRequirementsEntry
+                                {
+                                  Date = DateTime.MinValue
+                                }
+                            };
+            }
+
+            // skip the first header line, also skip #'s as these are comment lines
+
+            return File.ReadLines(file)
+                .Where(x => !x.StartsWith("#") && !string.IsNullOrWhiteSpace(x))
+                .Skip(1)
+                .Select(FromCsvLine)
+                .OrderBy(x => x.Date)
+                .ToArray();
+        }
+                
+        /// <summary>
+        /// Creates a new instance of <see cref="MarginRequirementsEntry"/> from the specified csv line
+        /// </summary>
+        /// <param name="csvLine">The csv line to be parsed</param>
+        /// <returns>A new <see cref="MarginRequirementsEntry"/> for the specified csv line</returns>
+        private MarginRequirementsEntry FromCsvLine(string csvLine)
+        {
+            var line = csvLine.Split(',');
+            var date = DateTime.MinValue;
+
+            if(!DateTime.TryParseExact(line[0], DateFormat.EightCharacter, CultureInfo.InvariantCulture, DateTimeStyles.None, out date))
+            {
+                Log.Trace("Couldn't parse date/time while reading future margin requirement file. Date {0}. Line: {1}", line[0], csvLine);
+            }
+
+            var initial = 0m;
+
+            if (!decimal.TryParse(line[1], out initial))
+            {
+                Log.Trace("Couldn't parse Initial margin requirements while reading future margin requirement file. Date {0}. Line: {1}", line[1], csvLine);
+            }
+
+            var maintenance = 0m;
+
+            if (!decimal.TryParse(line[2], out maintenance))
+            {
+                Log.Trace("Couldn't parse Maintenance margin requirements while reading future margin requirement file. Date {0}. Line: {1}", line[2], csvLine);
+            }
+
+            return new MarginRequirementsEntry()
+                    {
+                        Date = date,
+                        InitialOvernight = initial,
+                        MaintenanceOvernight = maintenance
+                    };
+        }
+
+
+        // Private POCO class for modeling margin requirements at given date
+        class MarginRequirementsEntry
```

Comment:
This looks like a similar pattern to the `MarketHoursDatabase` -- should we have a `MarginRequirementsDatabase` ? And then the constructor of this guy would accept an instance of the databae instead of hitting the disk directly